morphia/src/main/java/org/mongodb/morphia/AdvancedDatastore.java

```diff
@@ -257,51 +257,78 @@
     void setDecoderFact(DBDecoderFactory fact);
 
     /**
-     * Inserts an entity in to the named collection.
+     * Inserts an entity in to the mapped collection.
      *
-     * @param collection the collection to update
-     * @param entity     the entity to insert
-     * @param <T>        the type of the entity
+     * @param entity the entity to insert
+     * @param <T>    the type of the entity
      * @return the new key of the inserted entity
      */
-    <T> Key<T> insert(String collection, T entity);
+    <T> Key<T> insert(T entity);
 
     /**
      * Inserts an entity in to the mapped collection.
      *
      * @param entity the entity to insert
+     * @param wc     the WriteConcern to use when inserting
      * @param <T>    the type of the entity
      * @return the new key of the inserted entity
+     * @see WriteConcern
+     * @deprecated use {@link #insert(Object, InsertOptions)}
      */
-    <T> Key<T> insert(T entity);
+    @Deprecated
+    <T> Key<T> insert(T entity, WriteConcern wc);
 
     /**
      * Inserts an entity in to the mapped collection.
      *
-     * @param entity the entity to insert
-     * @param <T>    the type of the entity
+     * @param entity  the entity to insert
+     * @param options the options to apply to the save operation
+     * @param <T>     the type of the entity
      * @return the new key of the inserted entity
+     * @since 1.3
      */
-    <T> Iterable<Key<T>> insert(Iterable<T> entity);
+    <T> Key<T> insert(T entity, InsertOptions options);
```

Comment:
It looks them up when it can:  https://github.com/mongodb/morphia/blob/b94969f8eb6bd61592275a4f708ae580091cb543/morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java#L1261

There were a handful of gaps I found while auditing the other operations and I've filled them all in.